core/drivers/unix_socket.cc

```diff
@@ -38,53 +41,63 @@
 // TODO: Revise this once the interrupt mode is  implemented.
 
 #define RECV_SKIP_TICKS 256
-#define MAX_TX_FRAGS 8
+#define SIG_THREAD_EXIT SIGUSR2
 
-void UnixSocketPort::AcceptNewClient() {
-  int ret;
 
-  for (;;) {
-    ret = accept4(listen_fd_, nullptr, nullptr, SOCK_NONBLOCK);
-    if (ret >= 0) {
-      break;
-    }
+void UnixSocketPort::AcceptThread() {
+  sigset_t sigset;
+  sigfillset(&sigset);
+  sigdelset(&sigset, SIG_THREAD_EXIT);
 
-    if (errno != EINTR) {
-      PLOG(ERROR) << "[UnixSocket]:accept4()";
-    }
-  }
+  struct pollfd fds[2];
+  memset(fds, 0, sizeof(fds));
+  fds[0].fd = listen_fd_;
+  fds[0].events = POLLIN;
+  fds[1].events = POLLRDHUP;
 
-  recv_skip_cnt_ = 0;
+  while (true) {
+    fds[1].fd = client_fd_;
```

Comment:
Would you add a comment that negative file descriptors are ignored by `poll()` for clafirication?